File: src/dbflute/adf/meta/whiteXlsMandbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type WhiteXlsManDbm_T struct {
	df.BaseDBMeta
	ColumnXlsManId *df.ColumnInfo
	ColumnStringConverted *df.ColumnInfo
	ColumnTimestampZeroDefaultMillis *df.ColumnInfo
	ColumnTimestampZeroPrefixMillis *df.ColumnInfo
	ColumnTimestampZeroSuffixMillis *df.ColumnInfo
}

func (b *WhiteXlsManDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *WhiteXlsManDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var WhiteXlsManDbm *WhiteXlsManDbm_T

func Create_WhiteXlsManDbm() {
	WhiteXlsManDbm = new(WhiteXlsManDbm_T)
	WhiteXlsManDbm.TableDbName = "white_xls_man"
	WhiteXlsManDbm.TableDispName = "white_xls_man"
	WhiteXlsManDbm.TablePropertyName = "whiteXlsMan"
	WhiteXlsManDbm.TableSqlName = new(df.TableSqlName)
	WhiteXlsManDbm.TableSqlName.TableSqlName = "white_xls_man"
	WhiteXlsManDbm.TableSqlName.CorrespondingDbName = WhiteXlsManDbm.TableDbName

	var whiteXlsMan df.DBMeta
	whiteXlsMan = WhiteXlsManDbm
	WhiteXlsManDbm.DBMeta=&whiteXlsMan
	xlsManIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo xls_man_id
	xlsManIdSqlName.ColumnSqlName = "xls_man_id"
	xlsManIdSqlName.IrregularChar = false
	WhiteXlsManDbm.ColumnXlsManId = df.CCI(&whiteXlsMan, "xls_man_id", xlsManIdSqlName, "", "", "Long.class", "xlsManId", "", true, false,true, "int8", 19, 0, "",false,"","", "","","",false,"int64")
	stringConvertedSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo string_converted
	stringConvertedSqlName.ColumnSqlName = "string_converted"
	stringConvertedSqlName.IrregularChar = false
	WhiteXlsManDbm.ColumnStringConverted = df.CCI(&whiteXlsMan, "string_converted", stringConvertedSqlName, "", "", "String.class", "stringConverted", "", false, false,false, "varchar", 2147483647, 0, "",false,"","", "","","",false,"sql.NullString")
	timestampZeroDefaultMillisSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo timestamp_zero_default_millis
	timestampZeroDefaultMillisSqlName.ColumnSqlName = "timestamp_zero_default_millis"
	timestampZeroDefaultMillisSqlName.IrregularChar = false
	WhiteXlsManDbm.ColumnTimestampZeroDefaultMillis = df.CCI(&whiteXlsMan, "timestamp_zero_default_millis", timestampZeroDefaultMillisSqlName, "", "", "java.time.LocalDateTime.class", "timestampZeroDefaultMillis", "", false, false,false, "timestamp", 26, 3, "",false,"","", "","","",false,"df.NullTimestamp")
	timestampZeroPrefixMillisSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo timestamp_zero_prefix_millis
	timestampZeroPrefixMillisSqlName.ColumnSqlName = "timestamp_zero_prefix_millis"
	timestampZeroPrefixMillisSqlName.IrregularChar = false
	WhiteXlsManDbm.ColumnTimestampZeroPrefixMillis = df.CCI(&whiteXlsMan, "timestamp_zero_prefix_millis", timestampZeroPrefixMillisSqlName, "", "", "java.time.LocalDateTime.class", "timestampZeroPrefixMillis", "", false, false,false, "timestamp", 26, 3, "",false,"","", "","","",false,"df.NullTimestamp")
	timestampZeroSuffixMillisSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo timestamp_zero_suffix_millis
	timestampZeroSuffixMillisSqlName.ColumnSqlName = "timestamp_zero_suffix_millis"
	timestampZeroSuffixMillisSqlName.IrregularChar = false
	WhiteXlsManDbm.ColumnTimestampZeroSuffixMillis = df.CCI(&whiteXlsMan, "timestamp_zero_suffix_millis", timestampZeroSuffixMillisSqlName, "", "", "java.time.LocalDateTime.class", "timestampZeroSuffixMillis", "", false, false,false, "timestamp", 26, 3, "",false,"","", "","","",false,"df.NullTimestamp")

	WhiteXlsManDbm.ColumnInfoList = new(df.List)
	WhiteXlsManDbm.ColumnInfoList.Add(WhiteXlsManDbm.ColumnXlsManId)
	WhiteXlsManDbm.ColumnInfoList.Add(WhiteXlsManDbm.ColumnStringConverted)
	WhiteXlsManDbm.ColumnInfoList.Add(WhiteXlsManDbm.ColumnTimestampZeroDefaultMillis)
	WhiteXlsManDbm.ColumnInfoList.Add(WhiteXlsManDbm.ColumnTimestampZeroPrefixMillis)
	WhiteXlsManDbm.ColumnInfoList.Add(WhiteXlsManDbm.ColumnTimestampZeroSuffixMillis)


	WhiteXlsManDbm.ColumnInfoMap=make(map[string]int)
	WhiteXlsManDbm.ColumnInfoMap["xlsManId"]=0
		WhiteXlsManDbm.ColumnInfoMap["stringConverted"]=1
		WhiteXlsManDbm.ColumnInfoMap["timestampZeroDefaultMillis"]=2
		WhiteXlsManDbm.ColumnInfoMap["timestampZeroPrefixMillis"]=3
		WhiteXlsManDbm.ColumnInfoMap["timestampZeroSuffixMillis"]=4
	    WhiteXlsManDbm.PrimaryKey = true
    WhiteXlsManDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &whiteXlsMan
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(WhiteXlsManDbm.ColumnXlsManId)

	WhiteXlsManDbm.PrimaryInfo = new(df.PrimaryInfo)
	WhiteXlsManDbm.PrimaryInfo.UniqueInfo = ui
	
	var whiteXlsManMeta df.DBMeta = WhiteXlsManDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["WhiteXlsMan"] = &whiteXlsManMeta
}

File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/WhiteXlsManDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of white_xls_man. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class WhiteXlsManDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final WhiteXlsManDbm _instance = new WhiteXlsManDbm();
    private WhiteXlsManDbm() {}
    public static WhiteXlsManDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((WhiteXlsMan)et).getXlsManId(), (et, vl) -> ((WhiteXlsMan)et).setXlsManId(ctl(vl)), "xlsManId");
        setupEpg(_epgMap, et -> ((WhiteXlsMan)et).getStringConverted(), (et, vl) -> ((WhiteXlsMan)et).setStringConverted((String)vl), "stringConverted");
        setupEpg(_epgMap, et -> ((WhiteXlsMan)et).getTimestampZeroDefaultMillis(), (et, vl) -> ((WhiteXlsMan)et).setTimestampZeroDefaultMillis(ctldt(vl)), "timestampZeroDefaultMillis");
        setupEpg(_epgMap, et -> ((WhiteXlsMan)et).getTimestampZeroPrefixMillis(), (et, vl) -> ((WhiteXlsMan)et).setTimestampZeroPrefixMillis(ctldt(vl)), "timestampZeroPrefixMillis");
        setupEpg(_epgMap, et -> ((WhiteXlsMan)et).getTimestampZeroSuffixMillis(), (et, vl) -> ((WhiteXlsMan)et).setTimestampZeroSuffixMillis(ctldt(vl)), "timestampZeroSuffixMillis");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "white_xls_man";
    protected final String _tableDispName = "white_xls_man";
    protected final String _tablePropertyName = "whiteXlsMan";
    protected final TableSqlName _tableSqlName = new TableSqlName("white_xls_man", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnXlsManId = cci("xls_man_id", "xls_man_id", null, null, Long.class, "xlsManId", null, true, false, true, "int8", 19, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnStringConverted = cci("string_converted", "string_converted", null, null, String.class, "stringConverted", null, false, false, false, "varchar", 2147483647, 0, null, false, null, null, null, null, null, false);
//"sql.NullString"
    protected final ColumnInfo _columnTimestampZeroDefaultMillis = cci("timestamp_zero_default_millis", "timestamp_zero_default_millis", null, null, java.time.LocalDateTime.class, "timestampZeroDefaultMillis", null, false, false, false, "timestamp", 26, 3, null, false, null, null, null, null, null, false);
//"df.NullTimestamp"
    protected final ColumnInfo _columnTimestampZeroPrefixMillis = cci("timestamp_zero_prefix_millis", "timestamp_zero_prefix_millis", null, null, java.time.LocalDateTime.class, "timestampZeroPrefixMillis", null, false, false, false, "timestamp", 26, 3, null, false, null, null, null, null, null, false);
//"df.NullTimestamp"
    protected final ColumnInfo _columnTimestampZeroSuffixMillis = cci("timestamp_zero_suffix_millis", "timestamp_zero_suffix_millis", null, null, java.time.LocalDateTime.class, "timestampZeroSuffixMillis", null, false, false, false, "timestamp", 26, 3, null, false, null, null, null, null, null, false);
//"df.NullTimestamp"

    /**
     * xls_man_id: {PK, NotNull, int8(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnXlsManId() { return _columnXlsManId; }
    /**
     * string_converted: {varchar(2147483647)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnStringConverted() { return _columnStringConverted; }
    /**
     * timestamp_zero_default_millis: {timestamp(26, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnTimestampZeroDefaultMillis() { return _columnTimestampZeroDefaultMillis; }
    /**
     * timestamp_zero_prefix_millis: {timestamp(26, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnTimestampZeroPrefixMillis() { return _columnTimestampZeroPrefixMillis; }
    /**
     * timestamp_zero_suffix_millis: {timestamp(26, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnTimestampZeroSuffixMillis() { return _columnTimestampZeroSuffixMillis; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnXlsManId());
        ls.add(columnStringConverted());
        ls.add(columnTimestampZeroDefaultMillis());
        ls.add(columnTimestampZeroPrefixMillis());
        ls.add(columnTimestampZeroSuffixMillis());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnXlsManId()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.WhiteXlsMan"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.WhiteXlsManCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.WhiteXlsManBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<WhiteXlsMan> getEntityType() { return WhiteXlsMan.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public WhiteXlsMan newEntity() { return new WhiteXlsMan(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((WhiteXlsMan)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((WhiteXlsMan)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}
